functions/v2/test/index.test.js

```diff
@@ -13,94 +13,46 @@
 // limitations under the License.
 
 const assert = require('assert');
-const {request} = require('gaxios');
-const {exec} = require('child_process');
-const waitPort = require('wait-port');
+const {getTestServer} = require('@google-cloud/functions-framework/testing');
+const sinon = require('sinon');
+const supertest = require('supertest');
 
-const startFF = async (target, signature, port) => {
-  const ff = exec(
-    `npx functions-framework --target=${target} --signature-type=${signature} --port=${port}`
-  );
-  await waitPort({host: 'localhost', port});
-  return ff;
-};
+beforeEach(() => {
+  // require the module that includes the functions we are testing
+  require('../index');
```

Comment:
I feel like this needs more discussion.

If we shouldn't be running the FF during the testing process, that's fine - but we **do** need to clearly articulate (to customers) what the best practices are.

Reply:
It is articulated here: https://github.com/GoogleCloudPlatform/functions-framework-nodejs/blob/master/docs/testing-functions.md

Reply:
So I chatted with some of the other Node folks on my team, and we thought that an `exports.<method>` model (in the tested files) would be more idiomatic than `require`-ing a tested file directly.

Is there any reason we _shouldn't_ move to that model? As a reminder, it would look like this:

_index.js_
```
const someFunction = (cloudevent) => { ... }

// Expose function to external sources
functions.cloudEvent('someFunction', someFunction);
exports.someFunction = someFunction
```

Reply:
@matthewrobertson ah, that is new information!

Can/should we have the FF itself do an `export` under the hood?

If not, can/should we have the FF constructors return the function itself? (That way, we could do `exports.someFunction = functions.cloudevent('someFunction', ...)`)

_(Hopefully we're not going in circles here - I'd be more than happy to schedule a VC sync + create an agenda if you'd like me to.)_

Reply:
> Can/should we have the FF itself do an export under the hood, or (as you may have suggested elsewhere) some sort of functionsFramework.getFunction('someFunction') call?

Yes this exists now as a helper in the `/testing` module:

```
const {getFunction} = require('@google-cloud/functions-framework/testing');

describe('myFunc', () => {
  it('works, () => {
    const myFunc = getFunction('myFunc');
    // test myFunc...
  });
});
```

> I'd be more than happy to schedule a VC sync

sure happy to chat

Reply:
In this case, the module still needs to be required. The helper won't be able to find the function it is has never been loaded.

Reply:
Ah - so the helper does not handle the `require` internally.

Ideally we would add that, but I don't think it's worth blocking this PR. 🙂